### PR TITLE
Lock queues/topics until fully populated

### PR DESCRIPTION
**Pull request by brianmoran**

### CODE REVIEW

**Comment by p4tin** [comment]
This is good but I should have looked at other areas of code as well...  In the CreateQueue and createTopic function in sns/gosns.go and sqs/gosqs.go there are only read-locks while i mutate the list of queues or topics - we need to change that to read/wrote locks as well.

**Comment by brianmoran** [comment]
@p4tin I will look into that and update as well.

**Comment by brianmoran** [comment]
@p4tin - Let me know if I missed your point. Basically I replaced the **read** locks with Lock/Unlock

File: app/gosqs/gosqs.go [hunk 1]
```diff
@@ -437,7 +437,7 @@ func GetQueueUrl(w http.ResponseWriter, req *http.Request) {
 	//
 	//// Retrieve FormValues required
 	queueName := req.FormValue("QueueName")
-	SyncQueues.RLock()
+	SyncQueues.Lock()
```
Comment (p4tin) [inline app/gosqs/gosqs.go]: This one was not required cause the code is not changing anything just reading

File: app/gosqs/gosqs.go [hunk 2]
```diff
@@ -473,7 +473,7 @@ func GetQueueAttributes(w http.ResponseWriter, req *http.Request) {
 
 	log.Println("Get Queue Attributes:", queueName)
 	if queue, ok := SyncQueues.Queues[queueName]; ok {
-		SyncQueues.RLock()
+		SyncQueues.Lock()
```
Comment (p4tin) [inline app/gosqs/gosqs.go]: Same comment as above

**Comment by brianmoran** [comment]
@p4tin - Made the changes you suggested.

**Pull request merged**